Assets/HoloToolkit/Utilities/Scripts/Attributes/AnimationCurveDefaultAttribute.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using UnityEngine;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace HoloToolkit.Unity
+{
+    // Adds a 'default' button to an animation curve that will supply default curve values
+    [AttributeUsage(AttributeTargets.Field)]
+    public sealed class AnimationCurveDefaultAttribute : DrawOverrideAttribute
+    {
+        public WrapMode PostWrap { get; private set; }
+        public Keyframe StartVal { get; private set; }
+        public Keyframe EndVal { get; private set; }
+
+        public AnimationCurveDefaultAttribute(Keyframe startVal, Keyframe endVal, WrapMode postWrap = WrapMode.Loop)
+        {
+            PostWrap = postWrap;
+            StartVal = startVal;
+            EndVal = endVal;
+        }
+
+#if UNITY_EDITOR
+        public override void DrawEditor(UnityEngine.Object target, FieldInfo field, SerializedProperty property)
+        {
+            throw new NotImplementedException();
```

Comment:
Not Implemented?

Reply:
This attribute only works on fields. The attribute usage prohibits using them from being used on properties, so you'll get a warning if you try to use it incorrectly. Throwing a NotImplementedException just makes this limitation crystal-clear.